### PR TITLE
Remove python usage in the repo

### PR DESCRIPTION
**Pull request by jelmansouri**
Reducing the area of used languages since c++ has been used and the cc toolchain is a hard dependency for rust.
Only rust code and minimal cc remains to wrap executables when necessary

### CODE REVIEW

File: proto/optional_output_wrapper/bin.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
```
Comment (damienmg) [inline proto/optional_output_wrapper/bin.rs:1]:
```suggestion
// Copyright 2020 The Bazel Authors. All rights reserved.

```

File: proto/optional_output_wrapper/bin.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+fn ensure() -> Result<(),Box<dyn error::Error>> {
+	let index = args().position(|a| a=="--").ok_or("no --")?;
+	let optional_outputs = args().take(index).collect::<Vec<String>>();
+	let exe = args().nth(index+1).ok_or("no exe")?;
+	let exe_args = args().skip(index+2).collect::<Vec<String>>();
+	if exe_args.len() < 1 {
+		return Err("no exe args".into());
+	}
+	let _ = Command::new(exe).args(exe_args).output();
```
Comment (damienmg) [inline proto/optional_output_wrapper/bin.rs:33]: If I am reading this correctly this silently capture all the output, the legacy behavior was to simply forward stdout and stdin. Can this behavior be kept?

File: proto/optional_output_wrapper/BUILD [hunk 3]
```diff
@@ -0,0 +1,7 @@
+load("//rust:rust.bzl", "rust_binary")
```
Comment (damienmg) [inline proto/optional_output_wrapper/BUILD:1]: This is a simple 1 file binary, what is the logic for moving it to a new package?

Comment (jelmansouri) [reply proto/optional_output_wrapper/BUILD]: I just find it's find it's cleaner to have this in a separate directory with it's own build file.
Moved it to the same package as the parent.
Ideally we should move it to //proto/private I feel.

File: proto/optional_output_wrapper/bin.rs (hunk 2 above)
Comment (jelmansouri) [reply proto/optional_output_wrapper/bin.rs]: Yes you are correct, changed it to status, which inherits stdout, stderr & stdin handles

**Pull request merged**